### PR TITLE
Minor improvements to garage-push.

### PR DESCRIPTION
**Pull request by patrickvacek**
* Require credentials, no longer silently accept an undocumented default
option. Fix associated tests.
* Remove an unused getenv system call.
* More const-correctness and error checking.

Note that the return code from `curl_easy_setopt` is almost never checked. Is that worth changing?

### CODE REVIEW

**Comment by lbonn** [comment]
`curl_easy_setopt` has failure modes depending on which option (some never fail). The most usual case is failed allocation when it needs to copy a string.

We could use a quick wrapper that just throw a fatal exception in case of error.

**Comment by patrickvacek** [comment]
> We could use a quick wrapper that just throw a fatal exception in case of error.

Working on it.

File: src/sota_tools/ostree_object.cc [hunk 1]
```diff
@@ -235,15 +236,37 @@ void OSTreeObject::Upload(const TreehubServer &push_target, CURLM *curl_multi_ha
...
   assert(refcount_ > 0);  // At least our parent should have a reference to us
 
+  char *url = NULL;
+  curl_easy_getinfo(curl_handle_, CURLINFO_EFFECTIVE_URL, &url);
   long rescode = 0;  // NOLINT
   curl_easy_getinfo(curl_handle_, CURLINFO_RESPONSE_CODE, &rescode);
   if (current_operation_ == CurrentOp::kOstreeObjectPresenceCheck) {
-    if (rescode == 200) {
-      LOG_INFO << "Already Present:" << object_name_;
+    if (!url || !strstr(url, object_name_.c_str())) {
```
Comment (lbonn) [inline src/sota_tools/ostree_object.cc]: You're trying to see if the server sent us a redirect?

Comment (patrickvacek) [reply src/sota_tools/ostree_object.cc]: Nah, just verifying that the object the server found matches the object we have locally.

**Comment by patrickvacek** [comment]
I'm confused... nothing I changed should affect uptane_vector_tests, and indeed, when run the tests on master now, it breaks on the exact same tests. What happened?

**Comment by patrickvacek** [comment]
Never mind, my mistake. I accidentally changed the version of tuf-test-vectors.

**Comment by patrickvacek** [comment]
@lbonn Rebased and added comments. Is it enough?

**Pull request merged**